nixos/modules/profiles/macos-builder.nix

```diff
@@ -93,7 +93,12 @@ in
         };
       });
 
-  system.stateVersion = "22.05";
+  system = {
+    # To prevent gratuitous rebuilds on each change to Nixpkgs
+    nixos.revision = null;
+
+    stateVersion = "22.05";
```

Comment:
This is how the nixos tests set stateVersion, can we do the same here?

https://github.com/NixOS/nixpkgs/blob/e09e6d6f62314754f5ed578bd6ffa8e292a821fc/pkgs/test/nixos-functions/default.nix#L27

Reply:
(IMO, this should be a separate commit.)

Reply:
This config should be independent of stateVersion. It's within reach
 - #207313

I'll merge the uncontroversial bit, so that it can be made to work with `disabledModules` instead of the two more or less breaking changes in that PR.

Reply:
Here's how to avoid `stateVersion`
 - #207958

Reply:
Also, setting the `stateVersion` to match the current release is not necessarily the correct behavior here.  If you were to do that then any necessary migrations would never get applied if the user updates their `nixpkgs` pin in order to upgrade the builder.  The `stateVersion` is not supposed to change after the machine is deployed

Reply:
I agree, and I don't think a bare NixOS should depend on `stateVersion` for any such migrations, which is why I've opened #207958 as an alternative.